hibernate-core/src/main/java/org/hibernate/engine/jdbc/connections/spi/AbstractMultiTenantConnectionProvider.java

```diff
@@ -48,15 +48,17 @@ public boolean supportsAggressiveRelease() {
 
 	@Override
 	public boolean isUnwrappableAs(Class unwrapType) {
-		return ConnectionProvider.class.equals( unwrapType ) ||
-				MultiTenantConnectionProvider.class.equals( unwrapType ) ||
+		return MultiTenantConnectionProvider.class.equals( unwrapType ) ||
 				AbstractMultiTenantConnectionProvider.class.isAssignableFrom( unwrapType );
 	}
 
 	@Override
-	@SuppressWarnings( {"unchecked"})
+	@SuppressWarnings({"unchecked"})
 	public <T> T unwrap(Class<T> unwrapType) {
-		if ( isUnwrappableAs( unwrapType ) ) {
+		if ( ConnectionProvider.class.equals( unwrapType ) ) {
+			return (T) getAnyConnectionProvider();
```

Comment:
Unwrapping to anyConnectionProvider might be unintuitive in some use cases (maybe someone wants to get a tenant specific `ConnectionProvider`), so I would strongly suggest leaving the `unwrap` method as it was before the pull request. The testcase for HHH-11651 would also need to be adjusted to reflect the fact that unwrapping to a `ConnectionProvider` is not supported. `testUnwrappingConnectionProvider` should expect an `UnknownUnwrapTypeException` to be thrown.